typing/jkind.ml

```diff
@@ -2246,18 +2250,28 @@ let set_externality_upper_bound jk externality_upper_bound =
 let only_nullability = Axis_set.singleton (Nonmodal Nullability)
 
 let get_nullability ~jkind_of_type jk =
-  let ( ({ layout = _; mod_bounds; with_bounds = No_with_bounds } :
-          Allowance.right_only jkind_desc),
-        _ ) =
-    Layout_and_axes.normalize ~mode:Ignore_best ~jkind_of_type
-      ~map_type_info:(fun _ ti ->
-        { relevant_axes =
-            (* Optimization: We only care about the nullability axis *)
-            Axis_set.intersection ti.relevant_axes only_nullability
-        })
-      jk.jkind
+  (* Optimization: Usually, no with-bounds are relevant to nullability. If we check for
+     this case, we can avoid calling normalize. *)
+  let all_with_bounds_are_irrelevant =
+    jk.jkind.with_bounds |> With_bounds.to_seq
```

Comment:
We can probably even avoid the `to_seq` here by defining a `With_bounds.for_all`. But this is also good as-is.

Reply:
That shaved off enough to be worthwhile - down from 0.5% to 0.2% now.

Reply:
Inspired by this comment, I also did a similar change for `map_type_expr`, which also shaved off `0.7%`.